internal/access/passwordreset.go

```diff
@@ -28,13 +32,21 @@ func VerifiedPasswordReset(c *gin.Context, token, password string) (*models.Iden
 		}
 	}
 
-	credential, err := data.GetCredentialByUserID(tx, user.ID)
-	if err != nil {
+	if err := checkPasswordRequirements(tx, password); err != nil {
 		return nil, err
 	}
 
-	if err := checkPasswordRequirements(tx, password); err != nil {
-		return nil, err
+	credential, err := data.GetCredentialByUserID(tx, user.ID)
+	switch {
+	case errors.Is(err, internal.ErrNotFound):
+		if err := createCredential(tx, user, &models.Credential{}, password); err != nil {
```

Comment:
This seems potentially incorrect, if a user only exists in an identity provider this seems like it would allow them to add a password, in our current org controlled model this could be unexpected.

Reply:
You're talking about a password reset request for a user that exists in Infra but only through an IDP? That workflow will never reach this point because the password reset token is only created if an credential already exists for the user: https://github.com/infrahq/infra/blob/main/internal/server/passwordreset.go#L33